rqt_controller_manager/rqt_controller_manager/controller_manager.py

```diff
@@ -0,0 +1,425 @@
+        popup.resource_tree.expandAll()
+        popup.move(QCursor.pos())
+        popup.show()
+
+    def _on_header_menu(self, pos):
+        header = self._widget.table_view.horizontalHeader()
+
+        # Show context menu
+        menu = QMenu(self._widget.table_view)
+        action_toggle_auto_resize = menu.addAction('Toggle Auto-Resize')
+        action = menu.exec_(header.mapToGlobal(pos))
+
+        # Evaluate user action
+        if action is action_toggle_auto_resize:
+            if header.resizeMode(0) == QHeaderView.ResizeToContents:
+                header.setSectionResizeMode(QHeaderView.Interactive)
+            else:
+                header.setSectionResizeMode(QHeaderView.ResizeToContents)
+
+    def _activate_controller(self, name):
+        switch_controllers(
+            node=self._node,
+            controller_manager_name=self._cm_name,
+            deactivate_controllers=[],
+            activate_controllers=[name],
+            strict=SwitchController.Request.STRICT,
+            activate_asap=False,
+            timeout=0.3
+        )
+
+    def _deactivate_controller(self, name):
+        switch_controllers(
+            node=self._node,
+            controller_manager_name=self._cm_name,
+            deactivate_controllers=[name],
+            activate_controllers=[],
+            strict=SwitchController.Request.STRICT,
+            activate_asap=False,
+            timeout=0.3
+        )
+
+
+class ControllerTable(QAbstractTableModel):
+    """
+    Model containing controller information for tabular display.
+
+    The model allows display of basic read-only information like controller
+    name and state.
+    """
+
+    def __init__(self, controller_info,  icons, parent=None):
+        QAbstractTableModel.__init__(self, parent)
+        self._data = controller_info
+        self._icons = icons
+
+    def rowCount(self, parent):
+        return len(self._data)
+
+    def columnCount(self, parent):
+        return 2
+
+    def headerData(self, col, orientation, role):
+        if orientation == Qt.Horizontal and role == Qt.DisplayRole:
+            if col == 0:
+                return 'controller'
+            elif col == 1:
+                return 'state'
+        else:
+            return None
+
+    def data(self, index, role):
+        if not index.isValid():
+            return None
+
+        ctrl = self._data[index.row()]
+
+        if role == Qt.DisplayRole:
+            if index.column() == 0:
+                return ctrl.name
+            elif index.column() == 1:
+                if ctrl.state:
+                    return ctrl.state
+                else:
+                    return 'not loaded'
+
+        if role == Qt.DecorationRole:
+            if index.column() == 0:
+                if ctrl.state in self._icons:
+                    return self._icons[ctrl.state]
+                else:
+                    return None
```

Comment:
The next reduction assumes that if we don't have `ctrl.state` in `self._icons` we are going to retrieve a `None` without extra issues.
```suggestion
        if role == Qt.DecorationRole and index.column() == 0:
            return self._icons.get(ctrl.state)
```